drivers/gpio/gpio_emul.c

```diff
@@ -652,15 +665,31 @@ static int gpio_emul_pin_interrupt_configure(const struct device *port, gpio_pin
 
 	k_mutex_lock(&drv_data->mu, K_FOREVER);
 
+#ifdef CONFIG_GPIO_ENABLE_DISABLE_INTERRUPT
+	if (!(mode & GPIO_INT_ENABLE_DISABLE_ONLY)) {
+		drv_data->interrupts &= ~((gpio_port_pins_t)BIT(pin));
+	}
+#else
+	drv_data->interrupts &= ~((gpio_port_pins_t)BIT(pin));
+#endif /* CONFIG_GPIO_ENABLE_DISABLE_INTERRUPT */
+
 	switch (mode) {
 	case GPIO_INT_MODE_DISABLED:
 		drv_data->flags[pin] &= ~GPIO_EMUL_INT_BITMASK;
 		drv_data->flags[pin] |= GPIO_INT_DISABLE;
+#ifdef CONFIG_GPIO_ENABLE_DISABLE_INTERRUPT
+	case GPIO_INT_MODE_DISABLE_ONLY:
```

Comment:
You need to document fallthrough conditions.
```suggestion
	__fallthrough;
	case GPIO_INT_MODE_DISABLE_ONLY:
```

Reply:
Done